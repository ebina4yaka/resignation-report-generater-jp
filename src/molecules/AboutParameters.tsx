import React from 'react'
import Typography from '@material-ui/core/Typography'
import ParameterDescription from '../atoms/ParameterDescription'

export default function AboutParameters(): React.ReactElement {
  return (
    <div>
      <Typography
        variant="h5"
        align="center"
        color="textPrimary"
        paragraph
        gutterBottom
      >
        入力項目の説明
      </Typography>
      <div>
        <ParameterDescription name="退職者" description={['退職者の氏名']} />
        <ParameterDescription
          name="会社名"
          description={['退職する会社の社名']}
        />
        <ParameterDescription
          name="所属部署"
          description={['退職者の所属部署名']}
        />
        <ParameterDescription
          name="代表取締役"
          description={['退職する会社の代表者の氏名']}
        />
        <ParameterDescription
          name="退職理由"
          description={[
            '「○○○により～」の形式で出力されます。',
            '※○○○ = 入力内容',
          ]}
        />
        <ParameterDescription
          name="退職届日"
          description={['退職届を提出する日']}
        />
        <ParameterDescription
          name="消化有給休暇日数"
          description={['退職届を提出してから消化する有給休暇の日数']}
        />
        <ParameterDescription
          name="退職日"
          description={[
            '自動で退職届日から2週間後の日付が入ります。',
            '消化する有給休暇の日数が多い場合は調整してください。',
          ]}
        />
      </div>
    </div>
  )
}
