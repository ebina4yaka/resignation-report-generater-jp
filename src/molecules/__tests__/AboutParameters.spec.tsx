import React from 'react'
import renderer from 'react-test-renderer'
import AboutParameters from '../AboutParameters'

jest.mock(
  '../../atoms/ParameterDescription',
  () => 'MockedParameterDescription'
)

test('Snapshot test for AboutParameters', () => {
  const tree = renderer.create(<AboutParameters />).toJSON()
  expect(tree).toMatchSnapshot()
})
