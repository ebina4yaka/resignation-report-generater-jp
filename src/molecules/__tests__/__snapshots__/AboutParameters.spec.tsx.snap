// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for AboutParameters 1`] = `
<div>
  <p
    className="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextPrimary MuiTypography-gutterBottom MuiTypography-paragraph MuiTypography-alignCenter"
  >
    入力項目の説明
  </p>
  <div>
    <MockedParameterDescription
      description={
        Array [
          "退職者の氏名",
        ]
      }
      name="退職者"
    />
    <MockedParameterDescription
      description={
        Array [
          "退職する会社の社名",
        ]
      }
      name="会社名"
    />
    <MockedParameterDescription
      description={
        Array [
          "退職者の所属部署名",
        ]
      }
      name="所属部署"
    />
    <MockedParameterDescription
      description={
        Array [
          "退職する会社の代表者の氏名",
        ]
      }
      name="代表取締役"
    />
    <MockedParameterDescription
      description={
        Array [
          "「○○○により～」の形式で出力されます。",
          "※○○○ = 入力内容",
        ]
      }
      name="退職理由"
    />
    <MockedParameterDescription
      description={
        Array [
          "退職届を提出する日",
        ]
      }
      name="退職届日"
    />
    <MockedParameterDescription
      description={
        Array [
          "退職届を提出してから消化する有給休暇の日数",
        ]
      }
      name="消化有給休暇日数"
    />
    <MockedParameterDescription
      description={
        Array [
          "自動で退職届日から2週間後の日付が入ります。",
          "消化する有給休暇の日数が多い場合は調整してください。",
        ]
      }
      name="退職日"
    />
  </div>
</div>
`;
